Modules/_xxtestfuzz/fuzzer.c

```diff
@@ -411,8 +411,17 @@ int __lsan_is_turned_off(void) { return 1; }
 
 
 int LLVMFuzzerInitialize(int *argc, char ***argv) {
+    PyConfig config;
+    PyConfig_InitIsolatedConfig(&config);
+    PyStatus status;
     wchar_t* wide_program_name = Py_DecodeLocale(*argv[0], NULL);
-    Py_SetProgramName(wide_program_name);
+    status = PyConfig_SetString(&config, &config.program_name,
+                                wide_program_name);
+    if (PyStatus_Exception(status)) {
+        PyConfig_Clear(&config);
+        Py_ExitStatusException(status);
+    }
```

Comment:
You need to initialize Python with config, see example: https://docs.python.org/dev/c-api/init_config.html#initialization-with-pyconfig

You also need to call PyConfig_Clear() when you're done.

Reply:
If I understood correctly. You should remove the following code from LLVMFuzzerTestOneInput():

```
    if (!Py_IsInitialized()) {
        /* LLVMFuzzerTestOneInput is called repeatedly from the same process,
           with no separate initialization phase, sadly, so we need to
           initialize CPython ourselves on the first run. */
        Py_InitializeEx(0);
    }
```

And call ``Py_InitializeFromConfig()`` in LLVMFuzzerInitialize().

It seems like you need ``config.install_signal_handlers = 0;`` ;-)


Maybe keep an ``assert(Py_IsInitialized());`` in LLVMFuzzerTestOneInput().